bofire/utils/torch_tools.py

```diff
@@ -79,6 +80,44 @@ def get_linear_constraints(
     return constraints
 
 
+def get_nchoosek_constraints(domain: Domain) -> List[Callable[[Tensor], float]]:
+    """Transforms NChooseK constraints into a list of non-linear inequality constraint callables
+    that can be parsed by pydantic. For this purpose the NChooseK constraint is continuously
+    relaxed by countig the number of zeros in a candidate by a sum of narrow gaussians centered
+    at zero.
+
+    Args:
+        domain (Domain): Optimization problem definition.
+
+    Returns:
+        List[Callable[[Tensor], float]]: List of callables that can be used
+            as nonlinear equality constraints in botorch.
+    """
+
+    def narrow_gaussian(x, ell=1e-3):
+        return torch.exp(-0.5 * (x / ell) ** 2)
+
+    constraints = []
+    # ignore none also valid for the start
+    for c in domain.cnstrs.get(NChooseKConstraint):
+        assert isinstance(c, NChooseKConstraint)
```

Comment:
warum brauchst du das nochmal hier? Sollte doch eig bereits in constraints abgefangen sein

Reply:
Nur für pyright, damit der nicht meckert.